### PR TITLE
Use METADATA, not metadata.json

### PR DESCRIPTION
**Pull request by nathanlawrence-asana**
`metadata.json` doesn't exist for most wheels (PEP 0426 has been deferred), for example apache-airflow.  The older-style `METADATA` file should exist for all wheels.

This PR makes two changes to ease the transition:
1. It uses the `packaging` library (as opposed to `pkg_resources`).  `pkg_resources` in turn depends on `packaging`, but the `packaging` library is significantly cleaner and less crufty.
2. Because `packaging` the library conflicts with `packaging` the folder name in the `rules_python` repository, it's been renamed to `rules_python`.

### CODE REVIEW

File: rules_python/whl.py [hunk 1]
```diff
@@ -56,21 +57,25 @@ def _dist_info(self):
...
+        with whl.open(self._dist_info() + '/METADATA') as f:
+          # Why are we using email.parser?
+          #
+          # From PEP-0314:
+          #   The PKG-INFO file format is a single set of RFC-822 headers parseable by the rfc822.py module.
+          #   The field names listed in the following section are used as the header names.
+          #
+          # The rfc822.py module has been deprecated since version 2.3 in favor of the email package.
+          parser = email.parser.Parser()
+          self._metadata = parser.parsestr(f.read().decode('ascii', 'ignore'))
```
Comment (prebeta) [inline rules_python/whl.py:73]: I'm curious if there is a specific reason why rules_python does not want this workaround as well.

**Review by prebeta — APPROVED** [review]
LGTM

File: rules_python/whl.py (hunk 1 above)
Comment (jdavidheiser) [reply rules_python/whl.py:73]: I think they probably would want this workaround.  They have an open PR to fix it, but it's way jankier than what Nathan is doing here, so I hope we can contribute this back.

File: rules_python/whl.py [hunk 2]
```diff
@@ -56,21 +57,25 @@ def _dist_info(self):
...
-          pass
-      # fall back to METADATA file (https://www.python.org/dev/peps/pep-0427/)
-      with whl.open(self._dist_info() + '/METADATA') as f:
-        return self._parse_metadata(f.read().decode("utf-8"))
+    if self._metadata == None:
+      # Extract the structured data from METADATA in the WHL's dist-info
+      # directory.
+      with zipfile.ZipFile(self.path(), 'r') as whl:
+        with whl.open(self._dist_info() + '/METADATA') as f:
+          # Why are we using email.parser?
```
Comment (jdavidheiser) [inline rules_python/whl.py:65]: I was really skeptical of this, but the Pip source does the same thing.  Weird.

https://github.com/pypa/pip/blob/b7ed0445258bb18d8d2815e6208f9c996b5f347c/src/pip/_internal/utils/wheel.py#L180

**Review by jdavidheiser — APPROVED** [review]

File: rules_python/whl.py (hunk 2 above)
Comment (nathanlawrence-asana) [reply rules_python/whl.py:65]: That's why I left the comment 😄.  I went through the same level of skepticism.

**Pull request merged**